UIforETW/Utility.cpp

```diff
@@ -352,75 +743,112 @@ int DeleteFiles(HWND hwnd, const std::vector<std::wstring>& paths)
 		FOF_ALLOWUNDO | FOF_FILESONLY | FOF_NOCONFIRMATION,
 	};
 	// Delete using the recycle bin.
-	int result = SHFileOperation(&fileOp);
+	//TODO: IFileOperation?
+	const int result = ::SHFileOperationW(&fileOp);
 
 	return result;
 }
 
 void SetClipboardText(const std::wstring& text)

```

Comment:
I think that SetClipboardText and GetClipboardText and their associated helper functions are my least favorite changes in this pull request. They add a *ton* of code, they add an extra level of indirection to everything, they make the code harder to read (extra level of indirection) and it's not clear what value they add. The original functions shouldn't ever fail, but they had error checking just in case.

I think you understand the cognitive cost of increased lines of code, increased indirection (function calls instead of straightline code) and custom 'helper' functions that are non-standard and must be learned by anybody who wants to review the code. The original code was simple, handled most errors (could handle a couple more), and could be understood by any Win32 programmer with little effort. The new version is far more complicated and I still don't know what the added value is.